Hw 2

IrynaUkr opened the pull request:
homework 2

IrynaUkr committed:
initial

IrynaUkr committed:
homework 2 was added

IrynaUkr committed:
homework 2 was added

IrynaUkr committed:
git ignore was added

IrynaUkr committed:
Java code style was corrected

IrynaUkr committed:
empty line were deleted

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/Application.java:
```diff
@@ -0,0 +1,18 @@
+        String[] beanDefinitionNames = context.getBeanDefinitionNames();
+        Arrays.stream(beanDefinitionNames).forEach(System.out::println);
+        for (String beanName : beanDefinitionNames) {
+            System.out.println(context.getBean(beanName));
```
Бракує виконання пунку 12 на зразок:
`System.out.println(context.getBeanDefinition(beanName));`

IrynaUkr replied:
fixed

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/beans/BeanB.java:
```diff
@@ -0,0 +1,43 @@
+
+
+public class BeanB implements MyValidator {
+    @Value("${beanB.name}")
```
Якщо не оголошуєте бін @Component - тримайте такі анотації в конфігураційному класі

IrynaUkr replied:
fixed

KostyantynPanchenko reviewed (changes requested):
Доопрацюйте, будь ласка:

- усуньте `No newline at the end of file`
- усуньте інші зауваження
- перевірте форматування - є проблемні місця

IrynaUkr committed:
@Value annotation has been moved to the FirstConfig class;
Print configuration of beans was changed with using ApplicationContext context;
Mistake "new line at the end of file" was fixed.

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/config/FirstConfig.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Bean(value = "beanC", initMethod = "initC", destroyMethod = "destroyC")
+    @DependsOn(value = {"beanD", "beanB"})
+
```
Зайвий порожній рядок

IrynaUkr replied:
Дякую за вказівку, виправила.

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/config/FirstConfig.java:
```diff
@@ -0,0 +1,49 @@
+    @Bean(value = "beanC", initMethod = "initC", destroyMethod = "destroyC")
+    @DependsOn(value = {"beanD", "beanB"})
+
+    public BeanC beanC() {
```
Спосіб робочий, разом з тим, я б очікував більш елегантний і спеціалізований спосіб вставити значення із application.properties файлу, зокрема:

```java
@Bean(initMethod = "initMethod", destroyMethod = "destroyMethod")
@DependsOn("beanD")
public BeanB beanB(@Value("${beanB.name}") final String name, @Value("${beanB.value}") final int value) {
    return new BeanB(name, value);
}
```

IrynaUkr replied:
Виправила, так дійсно краще виглядає. Дякую за приклад використання.

KostyantynPanchenko commented on homework-2/src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+beanD.value=5
+
+beanC.name=nameC
+beanC.value=89
```
`No newline at the end of file`

IrynaUkr replied:
виправила

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/beans/BeanC.java:
```diff
@@ -0,0 +1,38 @@
+
+public class BeanC implements MyValidator {
+
+    private final String name;
```
Звернуть увагу на неконсистентність оголошення instance variable у класах BeanC, BeanD, BeanE - порожні рядки у різних місцях. Прошу привести код до однакового вигляду

IrynaUkr replied:
Дуже вдячна за докладний коментар. Не придавала значення форматуванню. Виправила.

KostyantynPanchenko commented on homework-2/src/main/java/com/epam/spring/homework2/Application.java:
```diff
@@ -0,0 +1,16 @@
+        Arrays.stream(context.getBeanDefinitionNames())
+                .forEach(name -> System.out.println(context.getBean(name)));
+        ((AnnotationConfigApplicationContext) context).close();
+    }
```
Бракує виконання пунку 12 на зразок:
`System.out.println(context.getBeanDefinition(beanName));`

IrynaUkr replied:
Дякую, додала.

IrynaUkr committed:
classes have been formatted to a consistent form;
method env.getProperty("beanName.name") was changed to the @Value annotation

IrynaUkr committed:
the printing beans definition was added
